### ISSUE TITLE
Proposal: ignore `no-param-reassign` for spread arguments.

### ISSUE THREAD

**Issue by nickclaw** [eslint/eslint#6218]
**What version of ESLint are you using?**
2.10.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```
{
    "parser": "babel-eslint",
    "extends": "airbnb",

    "ecmaFeatures": {
        "modules": true
    }
}

```

**What did you do? Please include the actual source code causing the issue.**
```js
export default function foo(...args) {
  args[0] = 'bar';
}
```

**What did you expect to happen?**
Since I believe the above code is equivalent to:
```js
export default function foo() {
  const args = Array(arguments.length);
  for (let i = 0; i < args.length; i++) { args[i] = arguments[i]; }
  args[0] = 'bar';
}
```

I think it would make sense to not treat this as reassigning a param, or at least have an option to ignore it.

**What actually happened? Please include the actual, raw output from ESLint.**
```
> eslint src


/path/to/index.js
  3:3  error  Assignment to property of function parameter 'args'  no-param-reassign

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
Could you try `no-param-reassign: ["error", {"props": false}]`?
AirBnB config has `{"props": true}` setting.

- http://eslint.org/docs/rules/no-param-reassign#props
- https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/best-practices.js#L152

**Comment by nickclaw** [comment]
Oh duh. That makes a lot of sense, I even saw that option but somehow didn't connect the dots. Thanks!
